lua/lspconfig/server_configurations/denols.lua

```diff
@@ -5,7 +5,11 @@ local function buf_cache(bufnr, client)
   local params = {}
   params['referrer'] = { uri = vim.uri_from_bufnr(bufnr) }
   params['uris'] = {}
-  client.request_sync('deno/cache', params)
+  client.request('deno/cache', params, function(err)
+    if err then
+      vim.api.nvim_err_writeln 'cache command failed'
```

Comment:
Is it necessary to add ctx relate information in there? because  you make it async then the buffer may changed.

Reply:
The context doesn't give real relevant information , for example

caching
```
import {} from "https://aze";
```
the context looks like this 
```
{
  bufnr = 1,
  client_id = 2,
  method = "deno/cache",
  params = {
    referrer = {
      uri = "file:///home/mrcool/dev/deno/lab/tmp/SZZ/a.ts"
    },
    uris = {}
  }
}
```
It doesn't give me the url I tried to cache, which is the only think I can see being useful to show (for example "errror caching `some-url`"

Reply:
if failed `local bufnr = vim.uri_to_bufnr(ctx.params.referrer.uri)` then prompt `cache for bufnr failed` .

Reply:
or `vim.uri_to_fname` prompt with a file name.

Reply:
ok added